pkg/ride/parserV2.go

```diff
@@ -0,0 +1,390 @@
+		if err != nil {
+			return nil, err
+		}
+		node, err := p.readDeclaration()
+		if err != nil {
+			return nil, err
+		}
+		fn, ok := node.(*FunctionDeclarationNode)
+		if !ok {
+			return nil, errors.Errorf("unexpected type of declaration %T", node)
+		}
+		fn.invocationParameter = invocationParameter
+		tree.Verifier = fn
+	}
+	tree.HasBlockV2 = p.seenBlockV2
+	tree.Digest = p.id
+	return tree, nil
+}
+
+func (p *parserV2) parseExpression() (*Tree, error) {
+	tree := &Tree{
+		contentType: p.header.content,
+		LibVersion:  int(p.header.library),
+	}
+	node, err := p.parseNext()
+	if err != nil {
+		return nil, err
+	}
+	tree.Verifier = node
+	tree.HasBlockV2 = p.seenBlockV2
+	tree.Digest = p.id
+	return tree, nil
+}
+
+func (p *parserV2) parseNext() (Node, error) {
+	t, err := p.r.ReadByte()
+	if err != nil {
+		return nil, err
+	}
+	switch t {
+	case tokenLong:
+		v, err := p.readLong()
+		if err != nil {
+			return nil, err
+		}
+		return NewLongNode(v), nil
+
+	case tokenBytes:
+		v, err := p.readBytes()
+		if err != nil {
+			return nil, err
+		}
+		return NewBytesNode(v), nil
+
+	case tokenString:
+		v, err := p.readString()
+		if err != nil {
+			return nil, err
+		}
+		return NewStringNode(v), nil
+
+	case tokenIf:
+		condition, err := p.parseNext()
+		if err != nil {
+			return nil, err
+		}
+		trueBranch, err := p.parseNext()
+		if err != nil {
+			return nil, err
+		}
+		falseBranch, err := p.parseNext()
+		if err != nil {
+			return nil, err
+		}
+		return NewConditionalNode(condition, trueBranch, falseBranch), nil
+
+	case tokenBlockV1:
+		name, err := p.readString()
+		if err != nil {
+			return nil, err
+		}
+		expr, err := p.parseNext()
+		if err != nil {
+			return nil, err
+		}
+		block, err := p.parseNext()
+		if err != nil {
+			return nil, err
+		}
+		return NewAssignmentNode(name, expr, block), nil
+
+	case tokenRef:
+		name, err := p.readString()
+		if err != nil {
+			return nil, err
+		}
+		return NewReferenceNode(name), nil
+
+	case tokenTrue:
+		return NewBooleanNode(true), nil
+
+	case tokenFalse:
+		return NewBooleanNode(false), nil
+
+	case tokenGetter:
+		object, err := p.parseNext()
+		if err != nil {
+			return nil, err
+		}
+		field, err := p.readString()
+		if err != nil {
+			return nil, err
+		}
+		return NewPropertyNode(field, object), nil
+
+	case tokenFunctionCall:
+		ft, err := p.r.ReadByte()
+		if err != nil {
+			return nil, err
+		}
+		function, err := p.readFunctionName(ft)
+		if err != nil {
+			return nil, err
+		}
+		argumentsCount, err := p.readInt()
+		if err != nil {
+			return nil, err
+		}
+		ac := int(argumentsCount)
+		arguments := make([]Node, ac)
+		for i := 0; i < ac; i++ {
+			arg, err := p.parseNext()
+			if err != nil {
+				return nil, err
+			}
+			arguments[i] = arg
+		}
+		return NewFunctionCallNode(function, arguments), nil
+
+	case tokenBlockV2:
+		p.seenBlockV2 = true
+		declaration, err := p.readDeclaration()
+		if err != nil {
+			return nil, err
+		}
```

Comment:
In `parserV1.go` I see this:
```go
declaration, err := p.readDeclaration()
if err != nil {
	return nil, err
}
if ad, ok := declaration.(*AssignmentNode); ok {
	ad.newBlock = true
}
```
But here I can't see this lines. Did you forget to add them?

Reply:
Yes, lost, will fix